reconcile/utils/smtp_client.py

```diff
@@ -23,14 +23,18 @@ def __init__(self, settings):
         self.passwd = smtp_config['password']
         self.mail_address = config['smtp']['mail_address']
 
-        self.client = smtplib.SMTP(host=self.host,
-                                   port=self.port)
-        self.client.send
-        self.client.starttls()
-        self.client.login(self.user, self.passwd)
-
+        self._client = None
         self._server = None
 
+    @property
+    def client(self):
+        if self._client is None:
+            self._client = smtplib.SMTP(host=self.host, port=self.port)
+            self._client.send
+            self._client.starttls()
+            self._client.login(self.user, self.passwd)
+        return self._client
```

Comment:
`@property` re-creates the client every time it is accessed - is this intended?
this is a version where the client is created once per class-instance and then cached:

```suggestion
    @functools.cached_property
    def client(self):
            client = smtplib.SMTP(host=self.host, port=self.port)
            client.send # <- btw, is this a side-effect hidden behind a property access? 🤨
            client.starttls()
            client.login(self.user, self.passwd)
        return client
```

https://docs.python.org/3/library/functools.html#functools.cached_property

Reply:
> @property re-creates the client every time it is accessed
can you expand on that?